lib/matplotlib/patches.py

```diff
@@ -137,9 +137,22 @@ def contains(self, mouseevent, radius=None):
         """
         if callable(self._contains):
             return self._contains(self, mouseevent)
+        inside = False
+        start = 0
         radius = self._process_radius(radius)
-        inside = self.get_path().contains_point(
-            (mouseevent.x, mouseevent.y), self.get_transform(), radius)
+        codes = self.get_path().codes
+        vertices = self.get_path().vertices
+        for i in range(1, len(codes)+1):
```

Comment:
can you check that we do not already have something in the path module that breaks a path up into separate paths?

Reply:
Yes, can we use `Path.iter_segments` here?

Reply:
Thanks you. I have tried the Path.iter_segments, it is not break the path to separate subpaths, but break it to every single vertice. I still need to do extra work to get the subpath(like create an empty vertices list to store each vertice or to find the index). Do you still want me to change to use Path.iter_segments?